File: app/src/main/java/org/michaelbel/moviemade/rest/api/service/COLLECTIONS.java
package org.michaelbel.moviemade.rest.api.service;

import org.michaelbel.moviemade.rest.model.v3.Collection;
import org.michaelbel.moviemade.rest.model.v3.CollectionImages;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface COLLECTIONS {

    @GET("collection/{collection_id}?")
    Observable<Collection> getDetails(
        @Path("collection_id") int id,
        @Query("api_key") String apiKey,
        @Query("language") String language
    );

    @GET("collection/{collection_id}/images?")
    Observable<CollectionImages> getImages(
        @Path("collection_id") int id,
        @Query("api_key") String apiKey,
        @Query("language") String language
    );
}

File: app/src/main/java/org/michaelbel/moviemade/data/service/COLLECTIONS.kt
package org.michaelbel.moviemade.data.service

import io.reactivex.Observable
import org.michaelbel.moviemade.data.entity.Collection
import org.michaelbel.moviemade.data.entity.ImagesResponse
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface COLLECTIONS {

    @GET("collection/{collection_id}?")
    fun getDetails(
        @Path("collection_id") id: Int,
        @Query("api_key") apiKey: String,
        @Query("language") language: String
    ): Observable<Collection>

    @GET("collection/{collection_id}/images?")
    fun getImages(
        @Path("collection_id") id: Int,
        @Query("api_key") apiKey: String,
        @Query("language") language: String
    ): Observable<ImagesResponse>

    // getTranslations
}